### PR TITLE
Refactor `ai/` to share code between service & scripts

### LINKED ISSUE #871: Refactor LLM library code
Right now we dupe the LLM code across ai/ft and ai/services and it's getting hard to maintain

### PR DESCRIPTION
**Pull request by wwwillchen**
Fixes #871.

This does a major refactoring of the code within `ai/` so that code, particularly `llm_lib.py` is properly shared between the online AI service and the offline scripts. This ensures consistency in calling LLMs and forming prompt context across the different workflows (e.g. generating dataset for eval, formatting goldens, and actual editor toolbar usage).

This removes a bunch of old scripts that we don't use anymore.

Finally, this uses `uv` so that Python dependencies/env are well-managed.

### CODE REVIEW

File: ai/README.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
+# AI package
+
+This contains all the AI-related code for Mesop. `ai/` is structured as an independent package and should have no dependencies on other parts of the codebase, nor dependendents from other parts of the codebase.
```
Comment (richard-to) [inline ai/README.md:3]: [typo] dependendents

File: ai/src/ai/common/llm_lib.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+from os import getenv
+from typing import NamedTuple
+
+from openai import OpenAI
+from openai.types.chat import (
+  ChatCompletionMessageParam,
+)
+
+SYSTEM_INSTRUCTION_PART_1_PATH = "src/ai/prompts/mesop_overview.txt"
+# SYSTEM_INSTRUCTION_PART_2_PATH = "src/ai/prompts/mini_docs.txt"
```
Comment (richard-to) [inline ai/src/ai/common/llm_lib.py:11]: [question] What's the commented out parts? Is part 2 the more extensive documentation?

File: ai/src/ai/common/llm_lib.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+SYSTEM_INSTRUCTION = SYSTEM_INSTRUCTION_PART_1  # + SYSTEM_INSTRUCTION_PART_2
+PROMPT_PATH = "src/ai/prompts/revise_prompt.txt"
+
+with open(PROMPT_PATH) as f:
+  REVISE_APP_BASE_PROMPT = f.read().strip()
+
+EDIT_HERE_MARKER = " # <--- EDIT HERE"
+
+
+class ApplyPatchResult(NamedTuple):
```
Comment (richard-to) [inline ai/src/ai/common/llm_lib.py:29]: [question] is there a reason for using NamedTuple here instead of a dataclass?

**Review by richard-to — APPROVED** [review]

File: ai/src/ai/common/llm_lib.py (hunk 2 above)
Comment (wwwillchen) [reply ai/src/ai/common/llm_lib.py:11]: yeah

**Pull request merged**